File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func modifiedList(nums []int, head *ListNode) *ListNode {
    var numSet = make(map[int]bool)
    for _, num := range nums {
        numSet[num] = true
    }
    
    var temp, prev = &ListNode{ -1, head, }, head
    head, prev = temp, temp
    temp = temp.Next
    for temp != nil {
        if numSet[temp.Val] {
            temp = temp.Next
            prev.Next = temp
        } else {
            temp = temp.Next
            prev = prev.Next
        }
    }
    return head.Next
}

File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        node = prev = ListNode(-1, head)
        curr = head
        val_set = set(nums)
        while(curr != None):
            if curr.val in val_set:
                prev.next = curr.next
                curr = prev.next
            else:
                prev = prev.next
                curr = curr.next
        
        return node.next